fvm/src/kernel/mod.rs

```diff
@@ -184,6 +185,10 @@ pub trait ActorOps {
 
     /// Returns the CodeCID for the supplied built-in actor type.
     fn get_code_cid_for_type(&self, typ: actor::builtin::Type) -> Result<Cid>;
+
+    /// Become changes the code of the running actor using the specified code cid;
+    /// it does not return unless there is an error
+    fn become_actor(&self, code_cid: &Cid) -> std::result::Result<(), Abort>;
```

Comment:
This should probably return `Result<BlockId>`.

- If it succeeds, we abort with the `BlockId` as the result.
- If it fails, we return a syscall error to the user.